includes/login-register.php

```diff
@@ -107,8 +107,19 @@ function edd_process_login_form( $data ) {
 		// Check for errors and redirect if none present.
 		$errors = edd_get_errors();
 		if ( ! $errors ) {
-			$redirect = apply_filters( 'edd_login_redirect', $data['edd_redirect'], 0 );
-			wp_redirect( $redirect );
+			$default_redirect_url = esc_url_raw( $data['edd_redirect'] );
+			if ( has_filter( 'edd_login_redirect' ) ) {
+				/**
+				 * Filters the redirect URL for EDD.
+				 *
+				 * @since 1.0
+				 * @param string $default_redirect_url  the URL to which to redirect
+				 * @param int                           deprecated as of 2.9.24; used to be user ID
+				 */
+				wp_redirect( apply_filters( 'edd_login_redirect', $default_redirect_url, 0 ) );
```

Comment:
```suggestion
				wp_redirect( apply_filters( 'edd_login_redirect', $default_redirect_url, $user->ID ) );
```

The `$user` object is still available further up so I see no reason we can't add that back in. (Docs will need to be updated further up.)